godel_surface_detection/src/services/blending_service_path_generation.cpp

```diff
@@ -57,83 +68,28 @@ filterPolygonBoundaries(const godel_process_path::PolygonBoundaryCollection& bou
   return filtered_boundaries;
 }
 
-bool SurfaceBlendingService::requestEdgePath(
-    const godel_process_path::PolygonBoundaryCollection& boundaries,
-    const geometry_msgs::Pose& boundary_pose,
-    const double z_ref,
-    visualization_msgs::Marker& visualization,
-    geometry_msgs::PoseArray& path)
-{
-  // Construct Visualization
-  geometry_msgs::Point pnt;
-  size_t idx(0);
-  for (godel_process_path::PolygonBoundaryCollection::const_iterator pb = boundaries.begin(), pb_end = boundaries.end(); pb != pb_end;
-       ++pb, ++idx)
-  {
-    for (godel_process_path::PolygonBoundary::const_iterator pt = pb->begin(), pt_end = pb->end(); pt != pt_end; ++pt)
-    {
-      pnt.x = pt->x;
-      pnt.y = pt->y;
-      pnt.z = z_ref;
-      visualization.points.push_back(pnt);
-    }
-  }
-
-  visualization.points.push_back(*(visualization.points.begin()));
 
-  // Add in an approach and depart vector
-  const geometry_msgs::Point& start_pt = visualization.points.front();
-  const geometry_msgs::Point& end_pt = visualization.points.back();
-  // Approach vector
-  std::vector<geometry_msgs::Point> approach_points;
-  for (std::size_t i = 0; i < SCAN_APPROACH_STEP_COUNT; ++i)
-  {
-    geometry_msgs::Point pt = start_pt;
-    pt.z += (SCAN_APPROACH_STEP_COUNT - i) * SCAN_APPROACH_STEP_DISTANCE;
-    approach_points.push_back(pt);
-  }
-  // Depart vector
-  std::vector<geometry_msgs::Point> depart_points;
-  for (std::size_t i = 0; i < SCAN_APPROACH_STEP_COUNT; ++i)
-  {
-    geometry_msgs::Point pt = end_pt;
-    pt.z += i * SCAN_APPROACH_STEP_DISTANCE;
-    depart_points.push_back(pt);
-  }
-  // Insert into path
-  visualization.points.insert(visualization.points.end(), depart_points.begin(), depart_points.end());
-  visualization.points.insert(visualization.points.begin(), approach_points.begin(), approach_points.end());
-
-  // Construct Poses
-  geometry_msgs::Pose p;
-  p.orientation.x = 0.0;
-  p.orientation.y = 0.0;
-  p.orientation.z = 0.0;
-  p.orientation.w = 1.0;
+bool SurfaceBlendingService::requestEdgePath(std::vector<pcl::IndicesPtr> &boundaries,
+                                             int index,
+                                             surfaceSegmentation& SS,
+                                             visualization_msgs::Marker& visualization,
+                                             geometry_msgs::PoseArray& path)
+{
+  geometry_msgs::Pose geo_pose;
+  std::vector<Eigen::Matrix4d, Eigen::aligned_allocator<Eigen::Matrix4d>> poses;
 
-  // Transform points to world frame and generate pose
-  Eigen::Affine3d boundary_pose_eigen;
-  Eigen::Affine3d eigen_p;
-  Eigen::Affine3d result;
+  // Get boundary trajectory and trim last two poses (last poses are susceptible to large velocity changes)
+  SS.getBoundaryTrajectory(boundaries, index, poses);
+  poses.resize(poses.size() - 2);
```

Comment:
We should assert that `poses.size()` is bigger than 2.

We should understand why the last points are wonky, if they are.